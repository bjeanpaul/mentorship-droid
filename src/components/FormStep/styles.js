import { merge } from 'lodash';
import { StyleSheet } from 'react-native';
import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';

import { DEVICE_HEIGHT, DEVICE_HEIGHT_MEDIUM } from 'src/constants/styles';


export default StyleSheet.create(merge({
  header: {
    margin: 18,
  },
  title: {
    margin: 7,
  },
  secondaryTitle: {
    fontSize: 14,
    fontFamily: FONT.REGULAR,
    fontWeight: FONT_WEIGHT.REGULAR,
    color: colors.TEXT_SECONDARY,
  },
  paginationContainer: {
    justifyContent: 'flex-end',
  },
  contentContainer: {
    flex: 1,
  },
}, DEVICE_HEIGHT > DEVICE_HEIGHT_MEDIUM && {
  header: {
    margin: 25,
  },
  title: {
    margin: 13,
  },
}));
