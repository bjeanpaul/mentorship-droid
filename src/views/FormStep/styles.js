import { StyleSheet } from 'react-native';


export default StyleSheet.create({
  title: {
    margin: 25,
    marginBottom: 50,
  },
  paginationContainer: {
    flex: 1,
    justifyContent: 'flex-end',
  },
});
