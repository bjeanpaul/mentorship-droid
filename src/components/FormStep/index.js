import React, { PropTypes } from 'react';
import { View } from 'react-native';
import { Text, Step, Pagination } from 'src/components';
import styles from './styles';

const FormStep = ({
  title,
  paginationDisabled,
  paginationBackDisabled,
  children,
  ...props,
}) => (
  <Step>
    <Text style={[Text.types.title, styles.title]}>{title}</Text>

    <View style={styles.contentContainer}>
      {children}
    </View>

    <View style={styles.paginationContainer}>
      <Pagination
        backDisabled={paginationBackDisabled}
        disabled={paginationDisabled}
        {...props}
      />
    </View>
  </Step>
);

FormStep.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any,
  paginationDisabled: PropTypes.bool,
  paginationBackDisabled: PropTypes.bool,
  onBackPress: PropTypes.func,
  onNextPress: PropTypes.func,
  onDonePress: PropTypes.func,
  last: PropTypes.bool,
};


export default FormStep;
