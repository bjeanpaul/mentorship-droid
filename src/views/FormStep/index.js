import React, { PropTypes } from 'react';
import { View } from 'react-native';
import { Text, Step, Pagination } from 'src/components';
import styles from './styles';

const FormStep = ({
  title,
  paginationDisabled,
  children,
  onBackPress,
  onNextPress,
}) => (
  <Step>
    <Text
      style={[Text.types.title, styles.title]}
    >
      {title}
    </Text>
    {children}
    <View style={styles.paginationContainer}>
      <Pagination
        onBackPress={onBackPress}
        onNextPress={onNextPress}
        disabled={paginationDisabled}
      />
    </View>
  </Step>
);

FormStep.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  paginationDisabled: PropTypes.bool,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};


export default FormStep;
