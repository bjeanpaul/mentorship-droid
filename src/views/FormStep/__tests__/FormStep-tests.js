import { noop } from 'lodash';
import React from 'react';
import { Text } from 'react-native';

import FormStep from 'src/views/FormStep';

describe('FormStep', () => {
  const createComponent = (props = {}) => (
    <FormStep
      title="Test Test Title"
      onBackPress={noop}
      onNextPress={noop}
      {...props}
    >
      <Text>I am a child.</Text>
    </FormStep>
  );

  it('should render', () => {
    const el = render(createComponent());
    expect(el).toMatchSnapshot();
  });
});
