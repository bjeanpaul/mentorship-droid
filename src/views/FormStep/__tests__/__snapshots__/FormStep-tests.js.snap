exports[`FormStep should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center"
        },
        Array [
          Object {
            "fontFamily": "Brandon Text_medium",
            "fontSize": 20,
            "fontWeight": "400"
          },
          Object {
            "margin": 25,
            "marginBottom": 50
          }
        ]
      ]
    }>
    Test Test Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    I am a child.
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end"
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 16
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300
            },
            undefined
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300
            },
            undefined
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
